### PR TITLE
Service discovery bug fixes

### PR DESCRIPTION
**Pull request by alena1108**
1) exportConfig - trim off "docker:" from the image name during export
2) don't let link services from 2 diff environments

### CODE REVIEW

**Comment by alena1108** [comment]
@cjellick @sonchang could you please review?

File: code/iaas/service-discovery/api/src/main/java/io/cattle/platform/servicediscovery/api/filter/ServiceAddRemoveLinkServiceValidationFilter.java [hunk 1]
```diff
@@ -41,21 +45,27 @@
...
         Long consumedServiceId = (Long) data.get(ServiceDiscoveryConstants.FIELD_SERVICE_ID);
         if (ACTIONS.get(action) == Operation.ADD) {
-            if (consumeMapDao.findNonRemovedMap(id, consumedServiceId) != null) {
+            Service service = objectManager.loadResource(Service.class, serviceId);
+            if (consumeMapDao.findNonRemovedMap(serviceId, consumedServiceId) != null) {
                 ValidationErrorCodes.throwValidationError(ValidationErrorCodes.NOT_UNIQUE,
                         ServiceDiscoveryConstants.FIELD_SERVICE_ID);
             }
+            Service consumedService = objectManager.loadResource(Service.class, consumedServiceId);
+            if (!consumedService.getEnvironmentId().equals(service.getEnvironmentId())) {
```
Comment (sonchang) [inline code/iaas/service-discovery/api/src/main/java/io/cattle/platform/servicediscovery/api/filter/ServiceAddRemoveLinkServiceValidationFilter.java]: We should add null checks for 'consumedService' and 'service'

File: tests/integration/cattletest/core/test_svc_discovery.py [hunk 2]
```diff
@@ -763,6 +763,37 @@ def test_remove_environment_w_active_svcs(super_client,
...
+    image_uuid = sim_context['imageUuid']
+    launch_config = {"imageUuid": image_uuid}
+
+    service1 = super_client.create_service(name=random_str(),
+                                           environmentId=env1.id,
+                                           networkId=nsp.networkId,
+                                           launchConfig=launch_config)
+    service1 = super_client.wait_success(service1)
+
+    env2 = admin_client.create_environment(name="compose")
```
Comment (sonchang) [inline tests/integration/cattletest/core/test_svc_discovery.py]: I wonder should we use a different name than env1?

Comment (alena1108) [inline tests/integration/cattletest/core/test_svc_discovery.py]: will add a random string

**Comment by sonchang** [comment]
LGTM

**Comment by cjellick** [comment]
LGTM

**Pull request merged**